### PR TITLE
fix: Make server compliant with Go SDK V2

### PR DESCRIPTION
**Pull request by bbernays**
Closes: #3 

The Go SDK V2 I think has more validation than other SDKs. This PR adds in the few headers/response codes required to get it working. 

If there is anything else that is needed for this to be approved and released please let me know!

### CODE REVIEW

**Comment by benkehoe** [comment]
Interesting, the docs don't mention the TTL header being reflected back in the token response. I think we should take this opportunity to add all the fields that the credentials call can return: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials

Add those fields (can we extract LastUpdated from the SDK? If not I think we can use the current time), change the header to be all lowercase (like the other headers referenced in the code), and add yourself to CONTRIBUTORS.md if you wish.

File: main.go [hunk 1]
```diff
@@ -258,6 +264,9 @@ func getTemporaryCredentials(awsConfig aws.Config) (Credentials, error) {
 		SecretAccessKey: *sessionCreds.Credentials.SecretAccessKey,
 		Token:           *sessionCreds.Credentials.SessionToken,
 		Expiration:      string(sessionExpiration),
+		Code:            "Success",
+		LastUpdated:     time.Now().UTC().String(),
+		Type:            "AWS-HMAC",
```
Comment (bbernays) [inline main.go:269]: This is what the value is in the documentation. Not sure if it should be something like `imds-credential-server`

**Comment by bbernays** [comment]
> Interesting, the docs don't mention the TTL header being reflected back in the token response. I think we should take this opportunity to add all the fields that the credentials call can return: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials
> 
Yeah, I added comments with references to the SDK code for future reference

> Add those fields (can we extract LastUpdated from the SDK? If not I think we can use the current time), change the header to be all lowercase (like the other headers referenced in the code), and add yourself to CONTRIBUTORS.md if you wish.

The there wasn't a `LastUpdated` coming from the SDK so I just used the current time

**Comment by avoidik** [comment]
I can confirm, it works. Thanks guys!

**Pull request merged**